store/src/main/java/com/nytimes/android/external/store3/base/impl/room/RoomInternalStore.java

```diff
@@ -0,0 +1,216 @@
+package com.nytimes.android.external.store3.base.impl.room;
+
+import com.nytimes.android.external.cache3.Cache;
+import com.nytimes.android.external.store3.base.Fetcher;
+import com.nytimes.android.external.store3.base.impl.CacheFactory;
+import com.nytimes.android.external.store3.base.impl.MemoryPolicy;
+import com.nytimes.android.external.store3.base.impl.StalePolicy;
+import com.nytimes.android.external.store3.base.impl.StoreUtil;
+import com.nytimes.android.external.store3.base.room.RoomPersister;
+
+import java.util.Collection;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ExecutionException;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import io.reactivex.Observable;
+import io.reactivex.disposables.Disposable;
+
+/**
+ * Store to be used for loading an object from different data sources
+ *
+ * @param <Raw>    data type before parsing, usually a String, Reader or BufferedSource
+ * @param <Parsed> data type after parsing
+ *                 <p>
+ */
+public class RoomInternalStore<Raw, Parsed, Key> implements RoomStore<Parsed, Key> {
```

Comment:
RoomInternalStore is the meat of it.  Notice that we don't need `stream` or `getRefreshing` as both of those can be achieved by Room without the need of the helpers.  Update a query that a RoomPersister is listening to and `get` will refresh on its own. If not preferable users can transform the `get` back into a `single`

Reply:
Most logic in this class stayed the same but changed to use Observable rather than Single